nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/test/java/org/apache/nifi/processors/solr/TestFetchSolr.java

```diff
@@ -0,0 +1,380 @@
+                reader.endObject();
+            }
+        }
+        reader.endObject();
+
+        JsonReader reader_stats = new JsonReader(new InputStreamReader(new ByteArrayInputStream(
+                runner.getContentAsByteArray(runner.getFlowFilesForRelationship(FetchSolr.STATS).get(0)))));
+        reader_stats.beginObject();
+        assertEquals(reader_stats.nextName(), "stats_fields");
+        reader_stats.beginObject();
+        assertFalse(reader_stats.hasNext());
+        reader_stats.endObject();
+        reader_stats.endObject();
+
+        reader.close();
+        reader_stats.close();
+    }
+
+    @Test
+    public void testStats() throws IOException {
+        final TestableProcessor proc = new TestableProcessor(solrClient);
+
+        TestRunner runner = TestRunners.newTestRunner(proc);
+        runner.setProperty(SolrUtils.SOLR_TYPE, SolrUtils.SOLR_TYPE_CLOUD.getValue());
+        runner.setProperty(SolrUtils.SOLR_LOCATION, "http://localhost:8443/solr");
+        runner.setProperty(SolrUtils.COLLECTION, "testCollection");
+        runner.setProperty(FetchSolr.SOLR_QUERY_STRING, "q=*:*&stats=true&stats.field=integer_single");
+        runner.enqueue(new ByteArrayInputStream("test".getBytes()));
+        runner.run();
+
+        runner.assertTransferCount(FetchSolr.STATS, 1);
+        JsonReader reader = new JsonReader(new InputStreamReader(new ByteArrayInputStream(
+                runner.getContentAsByteArray(runner.getFlowFilesForRelationship(FetchSolr.STATS).get(0)))));
+        reader.beginObject();
+        assertEquals(reader.nextName(), "stats_fields");
+        reader.beginObject();
+        assertEquals(reader.nextName(), "integer_single");
+        reader.beginObject();
+        while (reader.hasNext()) {
+            String name = reader.nextName();
+            switch (name) {
+                case "min": assertEquals(reader.nextString(), "0.0"); break;
+                case "max": assertEquals(reader.nextString(), "9.0"); break;
+                case "count": assertEquals(reader.nextInt(), 10); break;
+                case "sum": assertEquals(reader.nextString(), "45.0"); break;
+                default: reader.skipValue(); break;
+            }
+        }
+        reader.endObject();
+        reader.endObject();
+        reader.endObject();
+        reader.close();
+    }
+
+    @Test
+    public void testFailure() {
+        final TestableProcessor proc = new TestableProcessor(solrClient);
+
+        TestRunner runner = TestRunners.newTestRunner(proc);
+        runner.setProperty(SolrUtils.SOLR_TYPE, SolrUtils.SOLR_TYPE_CLOUD.getValue());
+        runner.setProperty(SolrUtils.SOLR_LOCATION, "http://localhost:8443/solr");
+        runner.setProperty(SolrUtils.COLLECTION, "testCollection");
+        runner.setProperty(FetchSolr.REQUEST_HANDLER, "/sel");
+
+        runner.setNonLoopConnection(false);
+        runner.run();
+
+        runner.assertTransferCount(FetchSolr.FAILURE, 1);

```

Comment:
There's an assertAllTransferred that can be used here to simply things.

Reply:
In this case, I wanted to validate that the flowfile actually is sent to FAILURE.